src/System.Private.CoreLib/src/Internal/Reflection/Core/NonPortable/RuntimeTypeUnifier.cs

```diff
@@ -43,11 +43,13 @@ internal static partial class RuntimeTypeUnifier
         //
         // Retrieves the unified Type object for given RuntimeTypeHandle (this is basically the Type.GetTypeFromHandle() api without the input validation.)
         //
-        public static Type GetTypeForRuntimeTypeHandle(RuntimeTypeHandle runtimeTypeHandle)
-            => RuntimeTypeHandleToTypeCache.Table.GetOrAdd(runtimeTypeHandle.RawValue);
-
         internal static Type GetRuntimeTypeForEEType(EETypePtr eeType)
-            => RuntimeTypeHandleToTypeCache.Table.GetOrAdd(eeType.RawValue);
+        {
+            // If writable data is supported, we shouldn't be using the hashtable - the runtime type
+            // is accessible through a couple indirections from the EETypePtr which is much faster.
+            Debug.Assert(Internal.Runtime.EEType.SupportsWritableData);
```

Comment:
```suggestion
            Debug.Assert(!Internal.Runtime.EEType.SupportsWritableData);
```

?

Reply:
Oops, good catch. I guess CppCodegen would quickly point that out.